### DISCUSSION TITLE
Using $this when not in object context

### DISCUSSION THREAD

**Discussion by Grzyb9k** [freekmurze/laravel-analytics#443] [Q&A]
Package not working for me. 

Laravel 8, PHP 7.4

![obraz](https://user-images.githubusercontent.com/73300122/154537970-0b8e5445-c3f5-49cd-a938-7fef64e2c463.png)

**Comment by azraf** [comment]
Same issue here. Can not perform any task for this issue.

**Comment by jfelisco** [comment]
double check spatie/laravel-analytics version. Version `4.0.0` and up doesn't support PHP 7.x

**Comment by AbiruzzamanMolla** [comment]
I am facing the same problem. Laravel v8.12 php7.4

**Comment by abdallahmaali** [comment]
same issues here, anyone fixed it?

**Comment by abdallahmaali** [comment]
i fixed these issues by the following 
upgrade PHP from7.4 to PHP 8.1 

in config/app.php
```

 'providers' => [
        Spatie\Analytics\AnalyticsServiceProvider::class,
]

'aliases' => [
        'Analytics' => Spatie\Analytics\AnalyticsFacade::class,
]

```

in Controller
```

use \Analytics;
use Spatie\Analytics\Period;


 public function index()
    {
        $analytics = \Analytics::getAnalyticsService();
       dd(analytics);
    } 

```
